client/job.lua

```diff
@@ -410,139 +399,109 @@ if config.useTarget then
-                onExit = outStashZone,
-                inside = insideStashZone,
+                onEnter = function()
+                    if QBX.PlayerData.job.onduty then
+                        lib.showTextUI(locale('text.pstash_button'))
+                    end
+                end,
+                onExit = function()
+                    lib.hideTextUI()
+                end,
+                inside = function()
+                    OnKeyPress(function()
+                        openStash(i)
+                    end)
+                end,
             })
         end
 
         for i = 1, #sharedConfig.locations.armory do
-            local function enteredArmoryZone()
-                if QBX.PlayerData.job.onduty then
-                    lib.showTextUI(locale('text.armory_button'))
-                end
-            end
-
-            local function outArmoryZone()
-                lib.hideTextUI()
-            end
-
-            local function insideArmoryZone()
-                OnKeyPress(function()
-                    openArmory(i)
-                end)
-            end
-
             lib.zones.box({
                 coords = sharedConfig.locations.armory[i].locations[1],
                 size = vec3(1, 1, 2),
                 rotation = -20,
                 debug = config.debugPoly,
-                onEnter = enteredArmoryZone,
-                onExit = outArmoryZone,
-                inside = insideArmoryZone,
+                onEnter = function()
+                    if QBX.PlayerData.job.onduty then
+                        lib.showTextUI(locale('text.armory_button'))
+                    end
+                end,
+                onExit = function()
+                    lib.hideTextUI()
+                end,
+                inside = function()
+                    OnKeyPress(function()
+                        openArmory(i)
+                    end)
+                end,
             })
         end
 
-        local function enteredRoofZone()
-            if QBX.PlayerData.job.onduty then
-                lib.showTextUI(locale('text.elevator_main'))
-            else
-                lib.showTextUI(locale('error.not_ems'))
-            end
-        end
-
-        local function outRoofZone()
-            lib.hideTextUI()
-        end
-
-        local function insideRoofZone()
-            OnKeyPress(teleportToMainElevator)
-        end
-
         lib.zones.box({
             coords = sharedConfig.locations.roof[1],
             size = vec3(1, 1, 2),
             rotation = -20,
             debug = config.debugPoly,
-            onEnter = enteredRoofZone,
-            onExit = outRoofZone,
-            inside = insideRoofZone,
+            onEnter = function()
+                if QBX.PlayerData.job.onduty then
+                    lib.showTextUI(locale('text.elevator_main'))
+                else
+                    lib.showTextUI(locale('error.not_ems'))
+                end
+            end,
```

Comment:
```suggestion
            onEnter = function()
                local label = QBX.PlayerData.job.onduty and 'text.elevator_main' or 'error.not_ems'
                lib.showTextUI(locale(label))
            end,
```
Same re: readability